### PR TITLE
WIP: Proposal controller tests

### PR DESCRIPTION
**Pull request by sonalkr132**
**#in_progress**

### CODE REVIEW

File: spec/controllers/proposal_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        end
+      end
+    end
+
+    describe 'PATCH #confirm' do
+      context 'confirmed successfully' do
+        context 'user have registered for the conference' do
+          before do
+            request.env['HTTP_REFERER'] = '/'
+            # change state of event to accepted
```
Comment (sonalkr132) [inline spec/controllers/proposal_controller_spec.rb]: hey, how do I change state of event from new to accepted? `event.accept!` says:
> ArgumentError: wrong number of arguments (0 for 1)

**Comment by differentreality** [comment]
@sonalkr132 sth like this? https://github.com/openSUSE/osem/blob/master/spec/models/ability_spec.rb#L28

File: app/controllers/proposal_controller.rb [hunk 2]
```diff
@@ -46,31 +46,28 @@ def create
...
-      render action: 'new'
-      return
+      render action: 'edit'
     end
-
-    redirect_to(conference_program_proposal_index_path(conference_id: @conference.short_title),
-                notice: 'Proposal was successfully updated.')
   end
 
   def destroy
```
Comment (sonalkr132) [inline app/controllers/proposal_controller.rb]: Shouldn't this be named `withdraw`? This method also making `DELETE` request for updating the record when it should make `PATCH` request, not very RESTful.

Comment (sonalkr132) [inline app/controllers/proposal_controller.rb]: To save the event in `destroy` method [we are doing](https://github.com/openSUSE/osem/pull/867/files#diff-8c13e9cfe1f5edf7d4923e633a08f790R84):
```
@event.save(validate: false)
```

When for reset (`withdraw`, `restart` etc), we are doing:
```
if @event.save
   # do something
else
  # do something else
end
```

I think we should be consistent. Should I change `destroy` save to `if else` block too?

Comment (hennevogel) [inline app/controllers/proposal_controller.rb]: `destroy -> withdraw` and `if else` sounds like a plan yes

**Comment by hennevogel** [comment]
Please rebase and fix your commit message. Thanks

**Comment by sonalkr132** [comment]
@hennevogel This PR won't be complete until #872 gets fixed. I am skipping tests here. If I add them now, I will have to change everything again.

**Comment by hennevogel** [comment]
Alright lets close this then

**Pull request closed without merging**